diff --git a/vendor/tree_sitter/grammars/gdscript/src/scanner.c b/vendor/tree_sitter/grammars/gdscript/src/scanner.c
index 4a6c6c5..9dbb20b 100644
--- a/vendor/tree_sitter/grammars/gdscript/src/scanner.c
+++ b/vendor/tree_sitter/grammars/gdscript/src/scanner.c
@@ -243,9 +243,6 @@ bool tree_sitter_gdscript_external_scanner_scan(void *payload, TSLexer *lexer,
 
     bool error_recovery_mode =
         valid_symbols[STRING_CONTENT] && valid_symbols[INDENT];
-    bool within_brackets = valid_symbols[CLOSE_BRACE] ||
-                           valid_symbols[CLOSE_PAREN] ||
-                           valid_symbols[CLOSE_BRACKET];
 
     if (valid_symbols[STRING_CONTENT] && scanner->delimiters->len > 0 &&
         !error_recovery_mode) {
